src/test/java/org/folio/rest/api/ItemEffectiveLocationTest.java

```diff
@@ -0,0 +1,376 @@
+import static org.junit.Assert.assertTrue;
+
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.folio.rest.jaxrs.model.Item;
+import org.folio.rest.jaxrs.model.Items;
+import org.folio.rest.support.IndividualResource;
+import org.folio.rest.support.Response;
+import org.folio.rest.support.ResponseHandler;
+import org.folio.rest.support.client.LoanTypesClient;
+import org.folio.rest.support.client.MaterialTypesClient;
+import org.folio.rest.support.http.InterfaceUrls;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Test cases to verify effectiveLocationId property calculation that implemented
+ * as two triggers for holdings_record and item tables (see itemEffectiveLocation.sql)
+ */
+public class ItemEffectiveLocationTest extends TestBaseWithInventoryUtil {
+  private static String journalMaterialTypeID;
+  private static String canCirculateLoanTypeID;
+  private static UUID mainLibraryLocationId;
+  private static UUID annexLibraryLocationId;
+  private static UUID onlineLocationId;
+  private static UUID secondFloorLocationId;
+
+  @BeforeClass
+  public static void beforeAny() throws Exception {
+    StorageTestSuite.deleteAll(itemsStorageUrl(""));
+    StorageTestSuite.deleteAll(holdingsStorageUrl(""));
+    StorageTestSuite.deleteAll(instancesStorageUrl(""));
+
+    StorageTestSuite.deleteAll(materialTypesStorageUrl(""));
+    StorageTestSuite.deleteAll(locationsStorageUrl(""));
+    StorageTestSuite.deleteAll(locLibraryStorageUrl(""));
+    StorageTestSuite.deleteAll(locCampusStorageUrl(""));
+    StorageTestSuite.deleteAll(locInstitutionStorageUrl(""));
+    StorageTestSuite.deleteAll(loanTypesStorageUrl(""));
+
+    MaterialTypesClient materialTypesClient = new MaterialTypesClient(client, materialTypesStorageUrl(""));
+    journalMaterialTypeID = materialTypesClient.create("journal");
+    canCirculateLoanTypeID = new LoanTypesClient(client, loanTypesStorageUrl("")).create("Can Circulate");
+
+    LocationsTest.createLocUnits(true);
+    mainLibraryLocationId = LocationsTest.createLocation(null, "Main Library (Item)", "It/M");
+    annexLibraryLocationId = LocationsTest.createLocation(null, "Annex Library (item)", "It/A");
+    onlineLocationId = LocationsTest.createLocation(null, "Online (item)", "It/O");
+    secondFloorLocationId = LocationsTest.createLocation(null, "Second Floor (item)", "It/SF");
+  }
+
+  @Before
+  public void beforeEach() {
+    StorageTestSuite.deleteAll(itemsStorageUrl(""));
+    StorageTestSuite.deleteAll(holdingsStorageUrl(""));
+    StorageTestSuite.deleteAll(instancesStorageUrl(""));
+  }
+
+  @After
+  public void checkIdsAfterEach() {
+    StorageTestSuite.checkForMismatchedIDs("item");
```

Comment:
We also have a holdings trigger, therefore we should also check for mismatchedIDs in "holdings_record".

Reply:
thanks @julianladisch, fixed.